server/master/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -1391,27 +1377,20 @@ public void run() {
       sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
     }
 
-    // Start the daemon to scan the replication table and make units of work
-    replicationWorkDriver = new ReplicationDriver(this);
-    replicationWorkDriver.start();
-
-    // Start the daemon to assign work to tservers to replicate to our peers
-    replicationWorkAssigner = new WorkDriver(this);
-    replicationWorkAssigner.start();
-
-    // Advertise that port we used so peers don't have to be told what it is
-    context.getZooReaderWriter().putPersistentData(
-        getZooKeeperRoot() + Constants.ZMASTER_REPLICATION_COORDINATOR_ADDR,
-        replAddress.address.toString().getBytes(UTF_8), NodeExistsPolicy.OVERWRITE);
-
-    // Register replication metrics
-    MasterMetricsFactory factory = new MasterMetricsFactory(getConfiguration(), this);
-    Metrics replicationMetrics = factory.createReplicationMetrics();
-    try {
-      replicationMetrics.register();
-    } catch (Exception e) {
-      log.error("Failed to register replication metrics", e);
-    }
+    // if the replication name is ever set, then start replication services
+    final AtomicReference<TServer> replServer = new AtomicReference<>();
+    SimpleTimer.getInstance(getConfiguration()).schedule(() -> {
+      try {
+        if (!getConfiguration().get(Property.REPLICATION_NAME).isEmpty()) {
```

Comment:
Could check if `replServer.get() == null` before checking config as this may be a much faster check and the timer has less to do after enabling.

Reply:
Fixed in 0016dce1a80f7c8f8552252f148ba77f94da2725.  Also did the same in Tserver

Reply:
Only concern with how this is written now is that it's subject to a race condition if we ever had multiple invocations of this "block" being executed concurrently (we could start two replication Thrift servers). I think `SimpleTimer` does not allow this to happen though.

Reply:
The race condition exists prior to this update, when we were just checking if the name was empty, because there is no synchronization.  But I think we are OK because the ScheduleExecutorService should prevent two concurrent threads from ever running for the same task.  The javadoc states:
"If any execution of this task
     * takes longer than its period, then subsequent executions
     * may start late, but will not concurrently execute."